testsuite/tests/typing-short-paths/errors.ml

```diff
@@ -34,3 +34,35 @@ Line 7, characters 9-18:
 Error: This expression has type "c"
        It has no method "bar"
 |}]
+
+type _ ty = Char : char ty
+type pair = Pair : 'a ty * 'a -> pair
+
+let f = function
+  | Pair (Char, x) -> x + 1
+[%%expect {|
+type _ ty = Char : char ty
+type pair = Pair : 'a ty * 'a -> pair
+Line 5, characters 22-23:
+5 |   | Pair (Char, x) -> x + 1
+                          ^
+Error: This expression has type "$Pair_'a"
+       but an expression was expected of type "int"
+|}]
```

Comment:
I would add a comment that we would like short-path to print `char` rather than `$Pair_'a` in this context.